### ISSUE TITLE
New static files not found on first request, but working on following requests.

### ISSUE THREAD

**Issue by yosifovdimitar** [phoenixframework/phoenix#3224]
### Environment

* Elixir version (elixir -v): Elixir 1.7.4
* Phoenix version (mix deps): 1.4.0 - 1.3 also
* NodeJS version (node -v): v8.10.0
* NPM version (npm -v): 5.6.0
* Operating system: Windows 10

### Expected behavior
After adding new file to priv/static/images it should be immediately available to domain/images/file.jpg requests.

### Actual behavior
When the new file is requested for the first time, it gives a 404 error in the browser and a "(Phoenix.Router.NoRouteError) no route found for GET /images/file.gif (appWeb.Router)" error on the server. A second request for the same file, makes the request work, for some reason. First request always fails, even if I wait 5 minutes before making a request for the new file, and following requests are successful.

**Comment by chrismccord** [comment]
if you recompile the app (kill the server and re-run phx.server), does the first request work? Thanks!

**Comment by yosifovdimitar** [comment]
Request is working when app is recompiled. 
I also found out that any request for a non-existing file, that results in an error, will make a following request for the new file to work.
If I add test2.jpg to the folder, request notthere.gif, results in error, then request test2.jpg, the request works, otherwise it fails.
I also want to mention that I reinstalled windows, changed the Antivirus, added folders to exceptions, disabled AV, reinstalled Elixir and so on.
The only thing that works so far is recompiling the app or making an error request or requesting the new file twice.
